src/algorithms/crypto/domain.go

```diff
@@ -0,0 +1,9 @@
+package crypto
+
+type DomainSeparationTag int
+
+const (
+	DomainSeparationTag_TicketProduction DomainSeparationTag = 1 + iota
+	DomainSeparationTag_ElectionProof
```

Comment:
Remove election proof (stale) and replace with post, will also need one for block signing

Reply:
> Remove election proof (stale) and replace with post

i.e., replace `ElectionProof` with `ElectionPoSt`?

Reply:
> will also need one for block signing

Will leave this for you to fill in in subsequent PR -- presumably many more tags will be needed (this is just one example), and the intent is for clients of the API to just add them here.

Reply:
Yes, it would be `PoSt` more generally since could be `electionPoSt` or `SurprisePoSt`.

Your second point sgtm